src/value/borrowed.rs

```diff
@@ -188,14 +188,14 @@ impl<'val> Struct for BorrowedStruct<'val> {
 pub enum BorrowedValue<'val> {
     Null(IonType),
     Integer(AnyInt),
-    Float(f64),
+    Float(&'val f64),
```

Comment:
I don't know if I was particularly clear about it, but it it is a small value, we shouldn't borrow it.